components/admin_console/plugin_management/plugin_management.jsx

```diff
@@ -1002,7 +1004,9 @@ export default class PluginManagement extends AdminSettings {
                                         type='file'
                                         accept='.gz'
                                         onChange={this.handleUpload}
-                                        disabled={!enableUploads || !this.state.enable || this.state.requirePluginSignature}
+
+                                        /* Using props values to make sure these are set on the server and not just locally */
+                                        disabled={!enableUploads || !enable || this.props.config.PluginSettings.RequirePluginSignature}
```

Comment:
Is this worth its own variable and avoid the condition check duplication ?

Reply:
Good idea! Fixed. 👍